File: OTCDerivativesCalculatorModule/Project_CSharp/XmlBuilder/XmlSerializedClass/fpml-business-events-5-4/TradeAmendmentContent.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Xml;

namespace FpmlSerializedCSharp
{
    public class TradeAmendmentContent : AbstractEvent
    {
        public TradeAmendmentContent(XmlNode xmlNode)
        : base(xmlNode)
        {
            XmlNode tradeNode = xmlNode.SelectSingleNode("trade");
            
            if (tradeNode != null)
            {
                if (tradeNode.Attributes["href"] != null || tradeNode.Attributes["id"] != null) 
                {
                    if (tradeNode.Attributes["id"] != null) 
                    {
                        tradeIDRef_ = tradeNode.Attributes["id"].Value;
                        Trade ob = new Trade(tradeNode);
                        IDManager.SetID(tradeIDRef_, ob);
                    }
                    else if (tradeNode.Attributes["href"] != null)
                    {
                        tradeIDRef_ = tradeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        trade_ = new Trade(tradeNode);
                    }
                }
                else
                {
                    trade_ = new Trade(tradeNode);
                }
            }
            
        
            XmlNode agreementDateNode = xmlNode.SelectSingleNode("agreementDate");
            
            if (agreementDateNode != null)
            {
                if (agreementDateNode.Attributes["href"] != null || agreementDateNode.Attributes["id"] != null) 
                {
                    if (agreementDateNode.Attributes["id"] != null) 
                    {
                        agreementDateIDRef_ = agreementDateNode.Attributes["id"].Value;
                        XsdTypeDate ob = new XsdTypeDate(agreementDateNode);
                        IDManager.SetID(agreementDateIDRef_, ob);
                    }
                    else if (agreementDateNode.Attributes["href"] != null)
                    {
                        agreementDateIDRef_ = agreementDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        agreementDate_ = new XsdTypeDate(agreementDateNode);
                    }
                }
                else
                {
                    agreementDate_ = new XsdTypeDate(agreementDateNode);
                }
            }
            
        
            XmlNode executionDateTimeNode = xmlNode.SelectSingleNode("executionDateTime");
            
            if (executionDateTimeNode != null)
            {
                if (executionDateTimeNode.Attributes["href"] != null || executionDateTimeNode.Attributes["id"] != null) 
                {
                    if (executionDateTimeNode.Attributes["id"] != null) 
                    {
                        executionDateTimeIDRef_ = executionDateTimeNode.Attributes["id"].Value;
                        ExecutionDateTime ob = new ExecutionDateTime(executionDateTimeNode);
                        IDManager.SetID(executionDateTimeIDRef_, ob);
                    }
                    else if (executionDateTimeNode.Attributes["href"] != null)
                    {
                        executionDateTimeIDRef_ = executionDateTimeNode.Attributes["href"].Value;
                    }
                    else
                    {
                        executionDateTime_ = new ExecutionDateTime(executionDateTimeNode);
                    }
                }
                else
                {
                    executionDateTime_ = new ExecutionDateTime(executionDateTimeNode);
                }
            }
            
        
            XmlNode effectiveDateNode = xmlNode.SelectSingleNode("effectiveDate");
            
            if (effectiveDateNode != null)
            {
                if (effectiveDateNode.Attributes["href"] != null || effectiveDateNode.Attributes["id"] != null) 
                {
                    if (effectiveDateNode.Attributes["id"] != null) 
                    {
                        effectiveDateIDRef_ = effectiveDateNode.Attributes["id"].Value;
                        XsdTypeDate ob = new XsdTypeDate(effectiveDateNode);
                        IDManager.SetID(effectiveDateIDRef_, ob);
                    }
                    else if (effectiveDateNode.Attributes["href"] != null)
                    {
                        effectiveDateIDRef_ = effectiveDateNode.Attributes["href"].Value;
                    }
                    else
                    {
                        effectiveDate_ = new XsdTypeDate(effectiveDateNode);
                    }
                }
                else
                {
                    effectiveDate_ = new XsdTypeDate(effectiveDateNode);
                }
            }
            
        
            XmlNodeList paymentNodeList = xmlNode.SelectNodes("payment");
            
            if (paymentNodeList != null)
            {
                this.payment_ = new List<Payment>();
                foreach (XmlNode item in paymentNodeList)
                {
                    if (item.Attributes["href"] != null || item.Attributes["id"] != null) 
                    {
                        if (item.Attributes["id"] != null) 
                        {
                            paymentIDRef_ = item.Attributes["id"].Value;
                            payment_.Add(new Payment(item));
                            IDManager.SetID(paymentIDRef_, payment_[payment_.Count - 1 ]);
                        }
                        else if (item.Attributes["href"] != null)
                        {
                            paymentIDRef_ = item.Attributes["href"].Value;
                        }
                        else
                        {
                        payment_.Add(new Payment(item));
                        }
                    }
                    else
                    {
                        payment_.Add(new Payment(item));
                    }
                }
            }
            
        
        }
        
    
        #region Trade_
        private Trade trade_;
        public Trade Trade_
        {
            get
            {
                if (this.trade_ != null)
                {
                    return this.trade_; 
                }
                else if (this.tradeIDRef_ != null)
                {
                    trade_ = IDManager.getID(tradeIDRef_) as Trade;
                    return this.trade_; 
                }
                else
                {
                      return this.trade_; 
                }
            }
            set
            {
                if (this.trade_ != value)
                {
                    this.trade_ = value;
                }
            }
        }
        #endregion
        
        public string tradeIDRef_ { get; set; }
        #region AgreementDate_
        private XsdTypeDate agreementDate_;
        public XsdTypeDate AgreementDate_
        {
            get
            {
                if (this.agreementDate_ != null)
                {
                    return this.agreementDate_; 
                }
                else if (this.agreementDateIDRef_ != null)
                {
                    agreementDate_ = IDManager.getID(agreementDateIDRef_) as XsdTypeDate;
                    return this.agreementDate_; 
                }
                else
                {
                      return this.agreementDate_; 
                }
            }
            set
            {
                if (this.agreementDate_ != value)
                {
                    this.agreementDate_ = value;
                }
            }
        }
        #endregion
        
        public string agreementDateIDRef_ { get; set; }
        #region ExecutionDateTime_
        private ExecutionDateTime executionDateTime_;
        public ExecutionDateTime ExecutionDateTime_
        {
            get
            {
                if (this.executionDateTime_ != null)
                {
                    return this.executionDateTime_; 
                }
                else if (this.executionDateTimeIDRef_ != null)
                {
                    executionDateTime_ = IDManager.getID(executionDateTimeIDRef_) as ExecutionDateTime;
                    return this.executionDateTime_; 
                }
                else
                {
                      return this.executionDateTime_; 
                }
            }
            set
            {
                if (this.executionDateTime_ != value)
                {
                    this.executionDateTime_ = value;
                }
            }
        }
        #endregion
        
        public string executionDateTimeIDRef_ { get; set; }
        #region EffectiveDate_
        private XsdTypeDate effectiveDate_;
        public XsdTypeDate EffectiveDate_
        {
            get
            {
                if (this.effectiveDate_ != null)
                {
                    return this.effectiveDate_; 
                }
                else if (this.effectiveDateIDRef_ != null)
                {
                    effectiveDate_ = IDManager.getID(effectiveDateIDRef_) as XsdTypeDate;
                    return this.effectiveDate_; 
                }
                else
                {
                      return this.effectiveDate_; 
                }
            }
            set
            {
                if (this.effectiveDate_ != value)
                {
                    this.effectiveDate_ = value;
                }
            }
        }
        #endregion
        
        public string effectiveDateIDRef_ { get; set; }
        #region Payment_
        private List<Payment> payment_;
        public List<Payment> Payment_
        {
            get
            {
                if (this.payment_ != null)
                {
                    return this.payment_; 
                }
                else if (this.paymentIDRef_ != null)
                {
                    return this.payment_; 
                }
                else
                {
                      return this.payment_; 
                }
            }
            set
            {
                if (this.payment_ != value)
                {
                    this.payment_ = value;
                }
            }
        }
        #endregion
        
        public string paymentIDRef_ { get; set; }
        
    
        
    
    }
    
}

File: OTCDerivativesCalculatorModule/Project_Cpp/lib_static/FpmlSerialized/GenClass/fpml-business-events-5-4/TradeAmendmentContent.cpp
// TradeAmendmentContent.cpp 
#include "TradeAmendmentContent.hpp"
#ifdef ConsolePrint
    #include <iostream>
#endif 

namespace FpmlSerialized {

TradeAmendmentContent::TradeAmendmentContent(TiXmlNode* xmlNode)
: AbstractEvent(xmlNode)
{
    #ifdef ConsolePrint
        std::string initialtap_ = FileManager::instance().tap_;
        FileManager::instance().tap_.append("   ");
    #endif 
   //tradeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* tradeNode = xmlNode->FirstChildElement("trade");

   if(tradeNode){tradeIsNull_ = false;}
   else{tradeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- tradeNode , address : " << tradeNode << std::endl;
   #endif
   if(tradeNode)
   {
      if(tradeNode->Attribute("href") || tradeNode->Attribute("id"))
      {
          if(tradeNode->Attribute("id"))
          {
             tradeIDRef_ = tradeNode->Attribute("id");
             trade_ = boost::shared_ptr<Trade>(new Trade(tradeNode));
             trade_->setID(tradeIDRef_);
             IDManager::instance().setID(tradeIDRef_,trade_);
          }
          else if(tradeNode->Attribute("href")) { tradeIDRef_ = tradeNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { trade_ = boost::shared_ptr<Trade>(new Trade(tradeNode));}
   }

   //agreementDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* agreementDateNode = xmlNode->FirstChildElement("agreementDate");

   if(agreementDateNode){agreementDateIsNull_ = false;}
   else{agreementDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- agreementDateNode , address : " << agreementDateNode << std::endl;
   #endif
   if(agreementDateNode)
   {
      if(agreementDateNode->Attribute("href") || agreementDateNode->Attribute("id"))
      {
          if(agreementDateNode->Attribute("id"))
          {
             agreementDateIDRef_ = agreementDateNode->Attribute("id");
             agreementDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(agreementDateNode));
             agreementDate_->setID(agreementDateIDRef_);
             IDManager::instance().setID(agreementDateIDRef_,agreementDate_);
          }
          else if(agreementDateNode->Attribute("href")) { agreementDateIDRef_ = agreementDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { agreementDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(agreementDateNode));}
   }

   //executionDateTimeNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* executionDateTimeNode = xmlNode->FirstChildElement("executionDateTime");

   if(executionDateTimeNode){executionDateTimeIsNull_ = false;}
   else{executionDateTimeIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- executionDateTimeNode , address : " << executionDateTimeNode << std::endl;
   #endif
   if(executionDateTimeNode)
   {
      if(executionDateTimeNode->Attribute("href") || executionDateTimeNode->Attribute("id"))
      {
          if(executionDateTimeNode->Attribute("id"))
          {
             executionDateTimeIDRef_ = executionDateTimeNode->Attribute("id");
             executionDateTime_ = boost::shared_ptr<ExecutionDateTime>(new ExecutionDateTime(executionDateTimeNode));
             executionDateTime_->setID(executionDateTimeIDRef_);
             IDManager::instance().setID(executionDateTimeIDRef_,executionDateTime_);
          }
          else if(executionDateTimeNode->Attribute("href")) { executionDateTimeIDRef_ = executionDateTimeNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { executionDateTime_ = boost::shared_ptr<ExecutionDateTime>(new ExecutionDateTime(executionDateTimeNode));}
   }

   //effectiveDateNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* effectiveDateNode = xmlNode->FirstChildElement("effectiveDate");

   if(effectiveDateNode){effectiveDateIsNull_ = false;}
   else{effectiveDateIsNull_ = true;}

   #ifdef ConsolePrint
      FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- effectiveDateNode , address : " << effectiveDateNode << std::endl;
   #endif
   if(effectiveDateNode)
   {
      if(effectiveDateNode->Attribute("href") || effectiveDateNode->Attribute("id"))
      {
          if(effectiveDateNode->Attribute("id"))
          {
             effectiveDateIDRef_ = effectiveDateNode->Attribute("id");
             effectiveDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(effectiveDateNode));
             effectiveDate_->setID(effectiveDateIDRef_);
             IDManager::instance().setID(effectiveDateIDRef_,effectiveDate_);
          }
          else if(effectiveDateNode->Attribute("href")) { effectiveDateIDRef_ = effectiveDateNode->Attribute("href");}
          else { QL_FAIL("id or href error"); }
      }
      else { effectiveDate_ = boost::shared_ptr<XsdTypeDate>(new XsdTypeDate(effectiveDateNode));}
   }

   //paymentNode ----------------------------------------------------------------------------------------------------------------------
   TiXmlElement* paymentNode = xmlNode->FirstChildElement("payment");

   if(paymentNode){paymentIsNull_ = false;}
   else{paymentIsNull_ = true;}

   if(paymentNode)
   {
      for(paymentNode; paymentNode; paymentNode = paymentNode->NextSiblingElement("payment")){
          #ifdef ConsolePrint
              FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- paymentNode , address : " << paymentNode << std::endl;
          #endif
          if(paymentNode->Attribute("href") || paymentNode->Attribute("id"))
          {
              if(paymentNode->Attribute("id"))
              {
                  paymentIDRef_ = paymentNode->Attribute("id");
                  payment_.push_back(boost::shared_ptr<Payment>(new Payment(paymentNode)));
                  payment_.back()->setID(paymentIDRef_);
                  IDManager::instance().setID(paymentIDRef_, payment_.back());
              }
              else if(paymentNode->Attribute("href")) { paymentIDRef_ = paymentNode->Attribute("href");}
              else { QL_FAIL("id or href error"); }
          }
          else { payment_.push_back(boost::shared_ptr<Payment>(new Payment(paymentNode)));}
      }
   }
   else {
       #ifdef ConsolePrint
           FileManager::instance().outFile_ << FileManager::instance().tap_.c_str() << "- paymentNode , address : " << paymentNode << std::endl;
       #endif
   }

    #ifdef ConsolePrint
        FileManager::instance().tap_ = initialtap_;
    #endif 
}
boost::shared_ptr<Trade> TradeAmendmentContent::getTrade()
{
   if(!this->tradeIsNull_){
        if(tradeIDRef_ != ""){
             return boost::shared_static_cast<Trade>(IDManager::instance().getID(tradeIDRef_));
        }else{
             return this->trade_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<Trade>();
   }
}
boost::shared_ptr<XsdTypeDate> TradeAmendmentContent::getAgreementDate()
{
   if(!this->agreementDateIsNull_){
        if(agreementDateIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDate>(IDManager::instance().getID(agreementDateIDRef_));
        }else{
             return this->agreementDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDate>();
   }
}
boost::shared_ptr<ExecutionDateTime> TradeAmendmentContent::getExecutionDateTime()
{
   if(!this->executionDateTimeIsNull_){
        if(executionDateTimeIDRef_ != ""){
             return boost::shared_static_cast<ExecutionDateTime>(IDManager::instance().getID(executionDateTimeIDRef_));
        }else{
             return this->executionDateTime_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<ExecutionDateTime>();
   }
}
boost::shared_ptr<XsdTypeDate> TradeAmendmentContent::getEffectiveDate()
{
   if(!this->effectiveDateIsNull_){
        if(effectiveDateIDRef_ != ""){
             return boost::shared_static_cast<XsdTypeDate>(IDManager::instance().getID(effectiveDateIDRef_));
        }else{
             return this->effectiveDate_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return boost::shared_ptr<XsdTypeDate>();
   }
}
std::vector<boost::shared_ptr<Payment>> TradeAmendmentContent::getPayment()
{
   if(!this->paymentIsNull_){
        if(paymentIDRef_ != ""){
             return this->payment_;
        }else{
             return this->payment_;
        }
   }else
   {
      QL_FAIL("null Ptr");
      return std::vector<boost::shared_ptr<Payment>>();
   }
}
}

